server/aws-lsp-codewhisperer/src/language-server/chat/chatController.ts

```diff
@@ -171,35 +208,84 @@ export class ChatController implements ChatHandlers {
         }
 
         this.#chatSessionManagementService.deleteSession(params.tabId)
-        this.#telemetryController.removeConversationId(params.tabId)
+        this.#telemetryController.removeConversation(params.tabId)
     }
 
     onQuickAction(params: QuickActionParams, _cancellationToken: CancellationToken) {
         switch (params.quickAction) {
             case QuickAction.Clear: {
                 const sessionResult = this.#chatSessionManagementService.getSession(params.tabId)
 
+                this.#telemetryController.emitChatMetric({
+                    name: ChatTelemetryEventName.RunCommand,
+                    data: {
+                        // TODO verify
+                        cwsprChatCommandType: params.quickAction,
+                    },
+                })
+
+                this.#telemetryController.removeConversation(params.tabId)
+
                 sessionResult.data?.clear()
 
                 return {}
             }
 
             case QuickAction.Help:
+                this.#telemetryController.emitChatMetric({
+                    name: ChatTelemetryEventName.RunCommand,
+                    data: {
+                        cwsprChatCommandType: params.quickAction,
+                    },
+                })
                 return {
                     messageId: uuid(),
                     body: HELP_MESSAGE,
                 }
+            default:
+                return {}
+        }
+    }
+
+    async #getTriggerContext(params: ChatParams, metric: Metric<CombinedConversationEvent>) {
+        const lastMessageTrigger = this.#telemetryController.getLastMessageTrigger(params.tabId)
+
+        let triggerContext: TriggerContext
+
+        // this is the only way we can detect a follow up action
```

Comment:
Can https://github.com/aws/language-servers/blob/main/server/aws-lsp-codewhisperer/src/language-server/chat/chatController.ts#L132 this listener help with this?

Reply:
I tried to, but the chat prompt message comes first now. It will also be harder to follow if it relies on orchestration of two different handlers